runtime/src/main/java/org/corfudb/runtime/LogReplicationServiceClient.java

```diff
@@ -0,0 +1,339 @@
+                DestinationInfoVal.class, null, TableOptions.builder().build());
+
+        register();
+    }
+
+    /**
+     * Adds client and model to LogReplicationRegistrationTable
+     */
+    private void register() {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    if (txn.isExists(LR_REGISTRY_TABLE_NAME, clientKey)) {
+                        log.warn("Client already registered \nclientKey: {} \nclientInfo: {}",
+                                clientKey, txn.getRecord(replicationRegistrationTable, clientKey).getPayload().toString());
+                    }
+                    else {
+                        txn.putRecord(replicationRegistrationTable, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
+                }
+                catch (Exception e) {
+                    log.error("Unable to register client {}, Error: {}", clientName, e);
+                }
+                return null;
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param destination Cluster which the logicalGroup should be replicated to
+     * @return true if destination was newly registered
+     */
+    public boolean addDestination(String logicalGroup, String destination) {
+        if (Strings.isNullOrEmpty(logicalGroup) || Strings.isNullOrEmpty(destination)) {
+            log.error("Invalid logicalGroup or destination");
+            return false;
+        }
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    DestinationInfoVal clientDestinations;
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        if (clientDestinations.getDestinationIdsList().contains(destination)) {
+                            log.warn("Destination already added");
```

Comment:
One point to note is on JVM restarts the client could be executing this logic to addDestination() and LR code in the server should not rely on this subscriber signal to start adding the destination.